### PR TITLE
Troubleshooting guide

### LINKED ISSUE #23: Actor troubleshooting doc
In the App Development section of our documentation, developers would benefit greatly from a "commonly seen errors" section.

This section could provide common errors encountered during wasmcloud actor development. Some that come to mind are:
1. Link not correctly configured from actor to provider
2. panic in actor module
3. JSON serialization failure (happens commonly when manually `call`ing an actor)
4. Suggestions for `Malformed response from actor`

A document like this would not attempt to be a comprehensive resource for all actor errors. What this would be useful for is to give developers a starting off point with common causes of these errors, and if they have not made use of the `logging` provider then a resource that points them to use that, as stdout is not viewable from an actor module.

### PR DESCRIPTION
**Pull request by brooksmtownsend**
fixes #23 

Draft PR for now as this is unfinished, but committing the beginnings of our troubleshooting documentation

### CODE REVIEW

File: content/app-dev/debugging/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Errors and Debugging"
+date: 2022-01-19T11:02:05+06:00
+weight: 9
+draft: false
+---
+
+We all like to think we can write perfect code. Unfortunately, this isn't always the case. When you get errors in a distributed system it can be difficult to track down the source of the problem; the goal of this guide will be to give you the tools to do this within wasmCloud.
```
Comment (autodidaddict) [inline content/app-dev/debugging/_index.en.md:8]: change `problem; the goal` to `problem. The goal`

**Review by autodidaddict — APPROVED** [review]
I don't see any reason to keep this open until it's complete or perfect. As long as the content that's here is good, then I think we're good publishing it. We already have some under construction parts on the wasmcloud.dev site so there's precedent for that.

**Comment by brooksmtownsend** [comment]
> I don't see any reason to keep this open until it's complete or perfect. As long as the content that's here is good, then I think we're good publishing it. We already have some under construction parts on the wasmcloud.dev site so there's precedent for that.

Yeah I'd like to add a few more things to this guide, but certainly won't keep it open until fully complete. Would like to cover the simple areas where claims are forgotten, links aren't established, etc

File: content/app-dev/debugging/actors.en.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+impl HttpServer for KvCounterActor {
+    async fn handle_request(&self, ctx: &Context, req: &HttpRequest) -> RpcResult<HttpResponse> {
+```
+This happens at the wasmCloud runtime level when a provider attempts to deliver a HTTPRequest. In this case, wherever you're making this request will return a `500` error, but regardless of the capability you will always be able to find the following in the host logs:
+```
+[error] Actor does not have proper capabilities to receive this invocation
+[error] Invocation failure: actor is missing capability claims
+```
+
+### Missing Link Definitions
```
[the remark this answers is not in the document]
Comment (brooksmtownsend) [reply content/app-dev/debugging/actors.en.md:81]: added a section

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**